adapters/repos/db/lsmkv/segment_group_compaction.go

```diff
@@ -547,3 +556,41 @@ func (sg *SegmentGroup) compactionFitsSizeLimit(left, right *segment) bool {
 	totalSize := left.size + right.size
 	return totalSize <= sg.maxSegmentSize
 }
+
+// Warning: waitForFlushingToComplete gives you no synchronization guarantees,
+// it mainly acts as a hint that a flushing has just completed, but it gives no
+// guarantees that a new flushing hasn't started yet. The idea is that it's
+// better to delay the attempt to obtain a a maintenance lock because it will
```

Comment:
**fix(spelling,non-blocking)**: one extra `a` in `obtain a a maintenance`

Reply:
fixed